beacon-chain/sync/service.go

```diff
@@ -272,19 +274,32 @@ func (s *Service) Start() {
 // Stop the regular sync service.
 func (s *Service) Stop() error {
 	defer func() {
+		s.cancel()
```

Comment:
should we cancel after the rateLimiter.free()? what happens if cancel already kills the rate limiter